nengo_gui/components/component.py

```diff
@@ -31,9 +35,9 @@ class Template(object):
     def __init__(self, *args, **kwargs):
         self.args = args
         self.kwargs = kwargs
-    def create(self, vizsim):
-        uid = vizsim.viz.get_uid(self)
-        c = self.cls(vizsim, vizsim.viz.config[self], uid,
+    def create(self, sim):
+        uid = '_uid_%d' % id(self)
+        c = self.cls(sim, sim.config[self], uid,
```

Comment:
It's defined in the individual Templates

https://github.com/nengo/nengo_gui/blob/master/nengo_gui/components/value.py#L44

That said, one of the things I'm very much looking forward to is that I'm pretty sure that now that this change has been made, I can get rid of the whole Template/Component thing, getting rid of a lot of magic and annoyance in this code.....